platforms/core-runtime/service-registry-impl/src/main/java/org/gradle/internal/service/DefaultServiceRegistry.java

```diff
@@ -1117,6 +1117,10 @@ private ConstructorService(DefaultServiceRegistry owner, ServiceAccessScope acce
                 throw new ServiceValidationException("Cannot register an interface for construction.");
             }
 
+            if (Modifier.isAbstract(implementationType.getModifiers())) {
+                throw new ServiceValidationException("Cannot register an abstract type for construction.");
```

Comment:
💅  It would be good to include the class name in the error message, so it's immediately clear which class causes the problem.

I realize other messages closeby don't do it, but I think they should. We don't have to change them in this PR.